### PR TITLE
Param fix

### LINKED ISSUE #3523: Parameter Dialog UX Issues
* When opening the New Parameter dialog it should auto focus on the keyword field.
* Pressing `Enter` should submit the dialog.

The two above are to support keyboard only workflow, where you would hit Cmd+P -> Enter keyword -> Hit Enter, and can continue writing your query.

* "This is the keyword that will be added..." string always being split into two rows: maybe we should just show the keyword (the `{{ ... }}` part) on the second row?

* There should be some indication that we saved the query after hitting OK (when we save).

### PR DESCRIPTION
**Pull request by ranbena**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix

## Description
Added features:
* auto focus on name input
* submit on enter
* name help only on date range. Changed help text:

screen shot 2019-03-05 at 11 49 33

## Related Tickets & Documents
Fixes #3523

### CODE REVIEW

File: client/app/components/EditParameterSettingsDialog.jsx [hunk 1]
```diff
@@ -98,23 +119,31 @@ function EditParameterSettingsDialog(props) {
...
       {...props.dialog.props}
       title={isNew ? 'Add Parameter' : param.name}
-      onOk={onConfirm}
-      okText={isNew ? 'Add Parameter' : null}
-      okButtonProps={{ disabled: !isFulfilled() }}
+      width={600}
+      footer={[(
+        <Button key="cancel" onClick={props.dialog.dismiss}>Cancel</Button>
+      ), (
+        <Button key="submit" htmlType="submit" disabled={!isFulfilled()} type="primary" form="paramForm">
```
Comment (gabrieldutra) [inline client/app/components/EditParameterSettingsDialog.jsx]: This is way better to handle Forms in Modals :rocket:

File: client/app/components/EditParameterSettingsDialog.jsx [hunk 2]
```diff
@@ -18,8 +20,16 @@ function getDefaultTitle(text) {
...
 }
 
-function NameInput({ name, onChange, existingNames, setValidation }) {
-  let helpText = `This is what will be added to your query editor {{ ${name} }}`;
+function isTypeDate(type) {
+  return startsWith('date', type) && !isTypeDateRange(type);
+}
+
+function isTypeDateRange(type) {
+  return startsWith('date-range', type);
```
Comment (arikfr) [inline client/app/components/EditParameterSettingsDialog.jsx]: What about date/time range?

Comment (ranbena) [reply client/app/components/EditParameterSettingsDialog.jsx]: 🤦‍♂️ Fixed with `/-range/.test(type);`

File: client/app/components/EditParameterSettingsDialog.jsx [hunk 3]
```diff
@@ -143,7 +172,7 @@ function EditParameterSettingsDialog(props) {
             <Option value="datetime-range-with-seconds">Date and Time Range (with seconds)</Option>
           </Select>
         </Form.Item>
-        {includes(['date', 'datetime-local', 'datetime-with-seconds'], param.type) && (
+        {isTypeDate(param.type) && (
```
Comment (arikfr) [inline client/app/components/EditParameterSettingsDialog.jsx]: I don't know why, but it works only for `date` type:

image

(notice that there is no checkbox)

File: client/app/components/EditParameterSettingsDialog.jsx [hunk 4]
```diff
@@ -18,8 +20,16 @@ function getDefaultTitle(text) {
...
 }
 
-function NameInput({ name, onChange, existingNames, setValidation }) {
-  let helpText = `This is what will be added to your query editor {{ ${name} }}`;
+function isTypeDate(type) {
+  return startsWith('date', type) && !isTypeDateRange(type);
+}
+
+function isTypeDateRange(type) {
+  return /-range/.test(type);
```
Comment (arikfr) [inline client/app/components/EditParameterSettingsDialog.jsx]: Doesn't seem to work at all 😳 

image

File: client/app/components/EditParameterSettingsDialog.jsx (hunk 3 above)
Comment (ranbena) [reply client/app/components/EditParameterSettingsDialog.jsx]: 😅 Mercy fire me plz.

Fixed https://github.com/getredash/redash/pull/3528/commits/7d6e0d8f8c86265aa9e06c481a3be4cbf5cd6f04

File: client/app/components/EditParameterSettingsDialog.jsx (hunk 4 above)
Comment (ranbena) [reply client/app/components/EditParameterSettingsDialog.jsx]: It's only for keyword - not title

Comment (arikfr) [reply client/app/components/EditParameterSettingsDialog.jsx]: 🤦‍♂️

**Pull request merged**